src/database/models/user_item.rs

```diff
@@ -68,16 +82,29 @@ impl User {
         if let Some(row) = result {
             Ok(Some(User {
                 id,
-                github_id: row.github_id,
+                github_id: if row.public_github || override_settings {
+                    row.github_id
+                } else {
+                    None
+                },
                 name: row.name,
-                email: row.email,
+                email: if override_settings { row.email } else { None },
                 avatar_url: row.avatar_url,
                 username: row.username,
                 bio: row.bio,
                 created: row.created,
                 role: row.role,
                 badges: Badges::from_bits(row.badges as u64)
                     .unwrap_or_default(),
+                settings: if override_settings {
+                    Some(UserSettings {
+                        public_github: row.public_github,
+                        theme: FrontendTheme::from_str(&row.theme),
+                        locale: row.locale,
+                    })
+                } else {
+                    None
+                },
```

Comment:
I'm not going to put this comment everywhere, But I think it's a lot better for code quality if these permissions checks happen in the actual conversion from a DBUser -> User. This stuff is repeated a lot and leaves the potential for when future calls are added for someone to forget putting the checks. If it's all in one place there's no problem